Draft 2024 schedule

drewbrew opened the pull request:
1. Adds a bunch of Python tooling from last year and adjusts it for the 2024 markdown formatting
2. Adds speakers for 2024
3. Adds initial (not final) schedule for 2024

mtrythall committed:
Adds room and difficulty

mtrythall committed:
Add spacer

mtrythall committed:
Merge branch 'main' into stub-schedule

mtrythall committed:
Merge branch 'main' into stub-schedule

mtrythall committed:
Merge branch 'main' into stub-schedule

mtrythall committed:
Use flexbox instead of grid for schedule

mtrythall committed:
Adds base graphics for social cards

mtrythall committed:
Merge branch 'main' into stub-schedule

mtrythall committed:
Fix CSS compilation issue

mtrythall committed:
Adds single card design

mtrythall committed:
Pull in talk

mtrythall committed:
More robust presenter social template

mtrythall committed:
Session social card

mtrythall committed:
Support 2 presenters

mtrythall committed:
Support 3 presenters

mtrythall committed:
Merge branch 'main' into stub-schedule

mtrythall committed:
Add schedule link

drewbrew committed:
🐍 Initial Python requirements

drewbrew committed:
🐍 Import models from 2023

drewbrew committed:
🐍 Add some basic constants to make coding easier

drewbrew committed:
🐍 update models based on @mtrythall's stub data

drewbrew committed:
Create manual schedule entries

drewbrew committed:
🐍 Export the manual schedule entries to YAML

drewbrew committed:
🐍 Ignore python cache

drewbrew committed:
🧑🏼‍🦲 Update speakers with 2024 data from pretalx

drewbrew committed:
🔧 Remove duplicate entry for Felipe

drewbrew committed:
Fix sprints day 2

drewbrew committed:
🐍 Fix mastodon handling and photo URL rename

drewbrew committed:
📅 Add first draft of 2023 schedule

Still needs keynotes and orientation but this is a start

drewbrew committed:
✏️ Tutorials are in-person

drewbrew committed:
🐍 Run black

mtrythall committed:
Fix primary nav

drewbrew committed:
✏️ Wipe 2023 speakers and regenerate 2024 speakers

Also copies the speaker blog post generator

mtrythall committed:
Fix sorting and grouping of sessions

drewbrew committed:
Merge remote-tracking branch 'origin/stub-schedule' into 2024-schedule

drewbrew committed:
Merge remote-tracking branch 'origin/main' into 2024-schedule

drewbrew committed:
📅 Add Sage's talk

drewbrew committed:
🗒️ Add instructions for importing from pretalx

drewbrew committed:
🗒️ Add instructions for making the speaker blog post

drewbrew committed:
Merge remote-tracking branch 'origin/main' into 2024-schedule

drewbrew committed:
📅 Add Vince's talk

drewbrew commented:
![localhost8082-schedule 2](https://github.com/djangocon/2024.djangocon.us/assets/7773256/18b9b45f-eb59-4839-bdf0-1c329b7da12a)

Draft schedule screenshot

Two known issues:
1. Times are off by an hour because my computer is in Central Time (@mtrythall is working on that)
2. The speakers' avatars aren't showing about halfway through in `shot-scraper` but seem fine in chrome (likely not something we need to care about)
![image](https://github.com/djangocon/2024.djangocon.us/assets/7773256/1c7833fb-4727-4a98-ad08-0d8b4e791fdb)

drewbrew committed:
🔧 Adjust the length of lunch to match the online talks

This way they show up on the same line

(Also adds t0 to the manual entries' tracks)

drewbrew commented:
![image](https://github.com/djangocon/2024.djangocon.us/assets/7773256/1fb65530-91ca-4118-887a-b2c60cbf2f4c)

Fixed the issue with lunch not showing up at the same line as the lunchtime online talk

drewbrew committed:
🗒️ Add a HHOWTO for mail-merging speaker times

drewbrew committed:
🔧 Hide the childcare blog post link until it's ready

drewbrew committed:
✏️ Edit readme to add how to generate manual.yaml

drewbrew committed:
🔧 Hide lunch/breakfast menus until they're ready

drewbrew committed:
🚜 Run black

mtrythall commented:
@drewbrew FWIW I think I posted this in Slack, but here were the fields I felt you could safely _remove_:

1. Slug
2. Published
3. Sitemap
4. Accepted
5. Group
6. Layout
7. Summary
8. Empty presenter photos (no photo: “”)
9. Abstract

Not sure if those are carrying over everywhere, but you have options.

drewbrew commented:
Awesome, thanks. I'll prune those in the morning.

mtrythall committed:
Fix dates on schedule

mtrythall committed:
Update datetimes on sessions

mtrythall commented:
This should have date fixes wrt timezones. I'd appreciate a check.

drewbrew committed:
🔧 Delete extra fields as suggested by @mtrythall

drewbrew committed:
🚜 Drop unneeded fields

Ref: https://github.com/djangocon/2024.djangocon.us/pull/37#issuecomment-2218860922

mtrythall reviewed (approved):
Did some testing with timezones. General display seems fine, and spot checking ordering made sense. Social cards render fine, and session pages still work.

I think this is good to go up, and any issues wouldn't be blocking.

drewbrew committed:
🚜 Add placeholders for opening remarks, etc

1. Opening remarks
2. Closing remarks
3. Keynotes
4. Lightning Talks
5. Panel

drewbrew commented:
I just added placeholders for opening remarks, keynotes, lightning talks, the panel, and closing remarks. It looks pretty solid. We are still waiting on one last speaker and then we can publish it.

drewbrew commented on src/_content/presenters/ryan-cheley.md:
```diff
@@ -0,0 +1,14 @@
+hidden: false
+name: Ryan Cheley
+permalink: /presenters/ryan-cheley/
+photo: ryan-cheley.jpeg
```
when I look at Ryan's picture on my local box, I see this: 
![image](https://github.com/user-attachments/assets/b0986314-5100-4005-85e8-b3f6fb3d449f)

That doesn't match the photo I'm adding here

drewbrew replied:
oh, that's because it's his 2023 picture that was in the stub branch: https://github.com/djangocon/2024.djangocon.us/blob/stub-schedule/src/_content/presenters/ryan-cheley.jpeg

How do I forcibly regenerate thumbnails locally?

drewbrew committed:
📅 Add placeholder schedule entries

drewbrew committed:
Fix track 1 detection

emullaney commented on src/_content/schedule/talks/2024-09-24-13-35-t1-lessons-from-e2e-testing-django-web-applications.md:
```diff
@@ -0,0 +1,19 @@
+room: Grand Ballroom III
+tags:
+- testing
+title: Lessons from E2E Testing Django Web Applications
```
Avindra asked that we change the title to "Lessons from E2E Testing Web Applications". @drewbrew can you make that change here?

jefftriplett replied:
@emullaney using the suggestion feature might be the easiest way to change it. 

```suggestion
title: Lessons from E2E Testing Web Applications
```

(holy moly this PR is so big it's hard to find/comment on things)

drewbrew committed:
Update talk title

Co-authored-by: Jeff Triplett <jefftriplett@users.noreply.github.com>

drewbrew committed:
🚜 Refresh data from pretalx

drewbrew committed:
📅 Add Patrick's talk and change the slug of Avindra's

emullaney commented on src/_content/schedule/talks/2024-09-24-13-35-t1-lessons-from-e2e-testing-web-applications.md:
```diff
@@ -0,0 +1,19 @@
+track: t1
+---
+
+Lots of companies are investing in end to end testing to release high quality software and remain competitive in today’s market. Now with libraries like Cypress and Playwright, end to end testing Django applications have become very intuitive and a whole lot of fun.  
```
can you change this from "Django applications" to "web applications"?

drewbrew replied:
```suggestion
Lots of companies are investing in end to end testing to release high quality software and remain competitive in today’s market. Now with libraries like Cypress and Playwright, end to end testing web applications have become very intuitive and a whole lot of fun.  
```

drewbrew committed:
✏️ Django -> web

drewbrew committed:
📅 Rearrange a few talks to avoid a Monday lunch conflict

This is not ideal having afternoon breaks at different times on
Monday/Tuesday, but if we have speakers drop due to visa issues,
we can rearrange.

drewbrew committed:
📢 Replace Rachell with Lacey as panel moderator

Rachell is organizing, while Lacey is moderating

drewbrew committed:
📅 Adjust Monday afternoon break to later

drewbrew committed:
📅 Shift panel start time back 10 minutes

drewbrew committed:
🖼️ Add Melanie's photo from 2022

drewbrew committed:
📅 Add more time for lightning talks

Shift every afternoon slot back slightly

drewbrew committed:
📅 Extend lightning talks by 10 minutes

drewbrew committed:
🔧 bring back tracks in manual schedule entries

drewbrew committed:
🕛 Adjust break times

jcjudkins committed:
blog post annoucing schedule

jcjudkins commented on src/_content/posts/announcing-schedule-dcus.md:
Need to update to a current image

tim-schilling replied:
I think that image is fine. /assets/img/blog/speaker-dcus2023.jpg is to an image of last year. I'm not sure how much current we can get.

tim-schilling replied:
Never mind. The new selection is good

jcjudkins committed:
blog img schedule

drewbrew committed:
Merge remote-tracking branch 'origin/main' into 2024-schedule

drewbrew committed:
Merge remote-tracking branch 'origin/2024-schedule' into 2024-schedule
